tools/ci-build.sh

```diff
@@ -94,17 +92,18 @@ P4C_PIP_PACKAGES="ipaddr \
                   scapy==2.4.5 \
                   clang-format>=15.0.4"
 
-
-if [[ "${DISTRIB_RELEASE}" == "18.04" ]] ; then
+if [[ "${DISTRIB_RELEASE}" == "18.04" ]] || [[ "$(which simple_switch 2> /dev/null)" != "" ]] ; then
   P4C_DEPS+=" libprotobuf-dev protobuf-compiler"
 else
+  apt-get update && apt-get install -y curl gnupg
   echo "deb http://download.opensuse.org/repositories/home:/p4lang/xUbuntu_${DISTRIB_RELEASE}/ /" | tee /etc/apt/sources.list.d/home:p4lang.list
   curl -L "http://download.opensuse.org/repositories/home:/p4lang/xUbuntu_${DISTRIB_RELEASE}/Release.key" | apt-key add -
+  apt-get update
```

Comment:
```Suggestion
```
` apt-get update` is no longer necessary here because there is one just after the "if" statement.